### ISSUE TITLE
conan search error message shows wrong reference

### ISSUE THREAD

**Issue by danimtb** [conan-io/conan#2580]
To help us debug your issue please explain:

- [x] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

Conan version 1.1.1

```
$ conan search paho-c/1.2.0@danimtb/testing -r conan-community
ERROR: Recipe not found: danimtb/paho-c/1.2.0/testing. [Remote: conan-community]
```

Reference in error message should be: `paho-c/1.2.0/@danimtb/testing`

**Comment by memsharded** [comment]
Yes, I think that the return should be a conan reference, same as the user input, not a path
